Support checkstyle report output #108. Support for the -r and -f para…

ngeor opened the pull request:
…meters of scss-lint.

This enables to set in options

    format: 'Checkstyle',
    require: 'scss_lint_reporter_checkstyle'

The names are in compliance with the CLI tool:

    Usage: scss-lint [options] [scss-files]
        -f, --format Formatter           Specify how to display lints
        -r, --require path               Require Ruby file

As a side-effect, with my fix you can use all built-in formatters of scss-lint (Config, Files, JSON, XML) as long as you use the reporterOutput option of grunt-scss-lint.

mariusbuescher commented:
Try setting the `options.timeout` configuration in Gruntfile.js to a higher value, e.g. `2500`. It seems other developers struggle with similar problems.

ngeor commented:
AppVeyor fails, Travis passes. If I go to AppVeyor history for this project, I can't really find any build that was successful. Is it possible that it is misconfigured?

ngeor commented:
I've updated the PR. It passes on Travis. On AppVeyor things are more difficult. I had to tell mocha-istanbul to hide the output of mocha + istanbul. AppVeyor gets confused and treats that output as an error. But even with this, AppVeyor dies at the final command, regarding code climate. That one however in the documentation says it only works for the master branch.

In the PR I've done some further changes:

- I've updated all gem and node dependencies to the latest, as the master is quite old.
- I've set AppVeyor to run also for node 4.4 (Travis picks up the latest anyway)
- Unit tests are written in various styles and most of them are integration tests anyway. I grouped them based on the technique that is being used.

At this point, I can't really proceed any further unless @ahmednuaman has a look at the PR and my findings regarding AppVeyor.

Also, I bumped up the version from 0.3.8 to 0.4.0 but perhaps it is worth bumping it up to 1.0.0 since all dependencies are upgraded.

ahmednuaman commented:
Let me remove this appveyor nonsense.

ahmednuaman commented:
Can you squash the commits please and hopefully that'll kick off the tests again.

ngeor commented:
Hm I don't see travis picking it up. I even added a dummy commit in .gitignore

ngeor commented:
Actually it works: https://travis-ci.org/ahmednuaman/grunt-scss-lint/builds/142527737 I don't know why it's not displayed inline in the PR in GitHub

ahmednuaman commented:
Can you rebase please?

ngeor commented:
Of course, should be fine now! Here's the latest build: https://travis-ci.org/ahmednuaman/grunt-scss-lint/builds/142999385

ahmednuaman commented on Gemfile.lock:
```diff
@@ -1,14 +1,20 @@
-    scss_lint (0.40.1)
-      rainbow (~> 2.0)
-      sass (~> 3.4.1)
+    rake (11.1.2)
```
Why rake?

ngeor replied:
It gets added when I run 'bundle'. Could be because of the upgrade I did on the Gemfile?

ahmednuaman replied:
I'm not sure if it's needed cos rake tasks aren't used here.

ngeor commented:
I took out the 'rake' dependency from Gemfile.lock . Latest build: https://travis-ci.org/ahmednuaman/grunt-scss-lint/builds/143670110

It fails with message: "Your Gemfile.lock is corrupt. The following gem is missing from the DEPENDENCIES
section: 'rake'"

ngeor commented:
I also removed the 'rake' nested dependency underneath scss-lint, same result. Like I said initially, I have not added the 'rake' dependency myself. In Gemfile, it is not present. This is apparently a dependency of scss-lint and it gets written into the generated file Gemfile.lock once one runs 'bundle'.

ngeor commented:
Thanks for merging!
